### PR TITLE
Add default values to some map API methods

### PR DESCRIPTION
**Pull request by rreusser**
This PR adds specific default values for some parameters, which I generally like, though I don't have particularly strong feelings. `style=null`, though true, is a slightly weird default since it puts things in a non-functional error-y state. Pulled from drive-by additions in https://github.com/mapbox/mapbox-gl-js/pull/10716.

I didn't add `container=null` since that's another which just throws an error, though maybe it's still true since the default state when unspecified *is* an error?

cc @domlet 

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [ ] include before/after visuals or gifs if this PR includes visual changes
 - [ ] write tests for all new functionality
 - [ ] document any changes to public APIs
 - [ ] post benchmark scores
 - [x] manually test the debug page
 - [ ] tagged `@mapbox/map-design-team` `@mapbox/static-apis` if this PR includes style spec API or visual changes
 - [ ] tagged `@mapbox/gl-native` if this PR includes shader changes or needs a native port
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'
 - [ ] add an entry inside this element for inclusion in the `mapbox-gl-js` changelog: `<changelog></changelog>`

### CODE REVIEW

**Comment by domlet** [comment]
Thank you @rreusser for improving docs consistency with these additions. 

I do favor presenting `container=null` like `style=null ` since these accurately document the default values.

I agree that it's a little weird, like _"Batteries not included"_ but I think it's appropriate to document. 👍🏼

**Review by domlet — APPROVED** [review]
I do favor adding `container=null` and I appreciate these updates!

**Comment by domlet** [comment]
Stretch goal – is there a test we can turn on to enforce requiring default values in the API docs?

**Comment by rreusser** [comment]
@domlet I've created a custom rule for this, but there are quite a few hits, even just in [`src/ui/map.js`](https://github.com/mapbox/mapbox-gl-js/blob/0eecef1d3fd42e9ebc41546e6ff1b292e9213491/src/ui/map.js):

```
 547:0:  @param {string} [position] position on the map to which the control will be added.
1216:0:  @param {PointLike|Array<PointLike>} [geometry] - The geometry of the query region in pixels:
1222:0:  @param {Object} [options] Options object.
1223:0:  @param {Array<string>} [options.layers] An array of [style layer IDs](https://docs.mapbox.com/mapbox-gl-js/style-spec/#layer-id) for the query to inspect.
1225:0:  @param {Array} [options.filter] A [filter](https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#filter)
1322:0:  @param {Object} [parameters] Options object.
1323:0:  @param {string} [parameters.sourceLayer] The name of the [source layer](https://docs.mapbox.com/help/glossary/source-layer/)
1325:0:  @param {Array} [parameters.filter] A [filter](https://docs.mapbox.com/mapbox-gl-js/style-spec/layers/#filter)
1369:0:  @param {Object} [options] Options object.
1844:0:  @param {string | Object} [layer.source] The data source for the layer.
1848:0:  @param {string} [layer.sourceLayer] (optional) The name of the [source layer](https://docs.mapbox.com/help/glossary/source-layer/) within the specified `layer.source` to use for this style layer.
1850:0:  @param {array} [layer.filter] (optional) An expression specifying conditions on source features.
1855:0:  @param {Object} [layer.paint] (optional) Paint properties for the layer.
1859:0:  @param {Object} [layer.layout] (optional) Layout properties for the layer.
1863:0:  @param {number} [layer.maxzoom] (optional) The maximum zoom level for the layer.
1867:0:  @param {number} [layer.minzoom] (optional) The minimum zoom level for the layer.
1871:0:  @param {Object} [layer.metadata] (optional) Arbitrary properties useful to track with the layer, but do not influence rendering.
1872:0:  @param {string} [layer.renderingMode] This is only applicable for layers with the type `custom`.
1874:0:  @param {string} [beforeId] The ID of an existing layer to insert the new layer before,
1948:0:  @param {string} [beforeId] The ID of an existing layer to insert the new layer before. When viewing the map, the `id` layer will appear beneath the `beforeId` layer. If `beforeId` is omitted, the layer will be appended to the end of the layers array and appear above all other layers on the map.
2034:0:  @param {Object} [options] Options object.
2075:0:  @param {Object} [options] Options object.
2106:0:  @param {Object} [options] Options object.
2133:0:  @param {Object} [options] Options object.
2238:0:  @param {ElevationQueryOptions} [options] options Object
2272:0:  @param {string} [feature.sourceLayer] (optional) *For vector tile sources, `sourceLayer` is required.*
2311:0:  @param {string} [feature.sourceLayer] (optional) *For vector tile sources, `sourceLayer` is required.*
2362:0:  @param {string} [feature.sourceLayer] (optional) *For vector tile sources, `sourceLayer` is required.*
```

My personal preference is to add this sort of linting since it enforces conventions and heads off a lot of small touch-up PRs for missed things like default parameters. But if not done well/thoughtfully, it can be worse than nothing at all. Would love to hear feedback about rules which would be beneficial for helping enforce doc quality.

**Comment by domlet** [comment]
@rreusser This is great, thank you for your work to scale and sustain this improvement!

> ...there are quite a few hits, even just in `src/ui/map.js`

I think our docs sprint offers us the opportunity to "face the noise" by resolving these instances of missing default values as a batch, so we can maintain a higher standard for completeness the future. 

> My personal preference is to add this sort of linting since it enforces conventions and heads off a lot of small touch-up PRs for missed things like default parameters. But if not done well/thoughtfully, it can be worse than nothing at all. 

If anyone on @mapbox/gl-js sees any opportunities to improve the implementation, let's discuss!

> Would love to hear feedback about rules which would be beneficial for helping enforce doc quality.

Here's a list of things the @mapbox/docs team would like to consistify in the GL JS API docs: https://github.com/mapbox/mapbox-gl-js-docs/issues/574

Thank you!

**Comment by domlet** [comment]
Discussed with @asheemmamoowala and we **will not** document `null` as the default value for either of these:

1. For style, we can say is it optional and explain that you can add a style this manually. 
2. For container, it should not appear optional and we should not document the default value as null.

I will make these changes to the branch.

File: src/ui/map.js [hunk 1]
```diff
@@ -182,12 +182,12 @@ const defaultOptions = {
...
- * @param {HTMLElement|string} options.container The HTML element in which Mapbox GL JS will render the map, or the element's string `id`. The specified element must have no children.
+ * @param {HTMLElement|string} [options.container=null] The HTML element in which Mapbox GL JS will render the map, or the element's string `id`. The specified element must have no children.
  * @param {number} [options.minZoom=0] The minimum zoom level of the map (0-24).
  * @param {number} [options.maxZoom=22] The maximum zoom level of the map (0-24).
  * @param {number} [options.minPitch=0] The minimum pitch of the map (0-85).
  * @param {number} [options.maxPitch=85] The maximum pitch of the map (0-85).
- * @param {Object|string} [options.style] The map's Mapbox style. This must be an a JSON object conforming to
+ * @param {Object|string} [options.style=null] The map's Mapbox style. This must be an a JSON object conforming to
  * the schema described in the [Mapbox Style Specification](https://mapbox.com/mapbox-gl-style-spec/), or a URL to
  * such JSON.
```
Comment (domlet) [inline src/ui/map.js:192]:
```suggestion
 * @param {Object|string} options.style The map's Mapbox style. This must be an a JSON object conforming to
 * the schema described in the [Mapbox Style Specification](https://mapbox.com/mapbox-gl-style-spec/), or a URL
 * to such JSON. Can accept a null value to allow adding a style manually.
```

**Review by domlet — APPROVED** [review]
I left two suggestions after a chat with @asheemmamoowala.

File: src/ui/map.js [hunk 2]
```diff
@@ -218,14 +219,14 @@ const defaultOptions = {
...
  * @param {string} [options.logoPosition='bottom-left'] A string representing the position of the Mapbox wordmark on the map. Valid options are `top-left`,`top-right`, `bottom-left`, `bottom-right`.
  * @param {boolean} [options.failIfMajorPerformanceCaveat=false] If `true`, map creation will fail if the performance of Mapbox GL JS would be dramatically worse than expected (a software renderer would be used).
  * @param {boolean} [options.preserveDrawingBuffer=false] If `true`, the map's canvas can be exported to a PNG using `map.getCanvas().toDataURL()`. This is `false` by default as a performance optimization.
- * @param {boolean} [options.antialias] If `true`, the gl context will be created with MSAA antialiasing, which can be useful for antialiasing custom layers. This is `false` by default as a performance optimization.
+ * @param {boolean} [options.antialias=false] If `true`, the gl context will be created with MSAA antialiasing, which can be useful for antialiasing custom layers. this is `false` by default as a performance optimization.
```
Comment (domlet) [inline src/ui/map.js:226]:
```suggestion
 * @param {boolean} [options.antialias=false] If `true`, the gl context will be created with MSAA antialiasing, which can be useful for antialiasing custom layers. This is `false` by default as a performance optimization.
```
capitalize first letter

**Comment by HeyStenson** [comment]
@rreusser is there anything else that needs to happen with this PR that I can help with? Or is it ready to be merged into the main testing branch?

**Comment by rreusser** [comment]
@HeyStenson re-running the tests (all, by accident) to get the green light, then it's good to go!

**Pull request merged**